pints/tests/test_slice_stepout.py

```diff
@@ -0,0 +1,397 @@
+        self.assertEqual(mcmc._fx_l, None)
+        self.assertEqual(mcmc._fx_r, None)
+
+        self.assertEqual(mcmc._m, 50)
+        self.assertEqual(mcmc._active_param_index, 0)
+
+    def test_first_run(self):
+        """
+        Tests the very first run of the sampler.
+        """
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([2., 4.])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Ask should fail if _ready_for_tell flag is True
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = True
+            mcmc.ask()
+
+        # Undo changes
+        mcmc._ready_for_tell = False
+
+        # Check whether _running flag becomes True when ask() is called
+        # Check whether first iteration of ask() returns x0
+        self.assertFalse(mcmc._running)
+        self.assertTrue(np.all(mcmc.ask() == x0))
+        self.assertTrue(mcmc._running)
+        self.assertTrue(mcmc._ready_for_tell)
+
+        # Tell should fail when log pdf of x0 is infinite
+        with self.assertRaises(ValueError):
+            fx = np.inf
+            mcmc.tell(fx)
+
+        # Calculate log pdf for x0
+        fx = log_pdf.evaluateS1(x0)[0]
+
+        # Tell should fail when _ready_for_tell is False
+        with self.assertRaises(RuntimeError):
+            mcmc._ready_for_tell = False
+            mcmc.tell(fx)
+
+        # Undo changes
+        mcmc._ready_for_tell = True
+
+        # Test first iteration of tell(). The first point in the chain
+        # should be x0
+        self.assertTrue(np.all(mcmc.tell(fx) == x0))
+
+        # We update the current sample
+        self.assertTrue(np.all(mcmc._current == x0))
+        self.assertTrue(np.all(mcmc._current == mcmc._proposed))
+
+        # We update the _current_log_pdf value used to generate the new slice
+        self.assertEqual(mcmc._current_log_pdf, fx)
+
+        # Check that the new slice has been constructed appropriately
+        self.assertTrue(mcmc._current_log_y < mcmc._current_log_pdf)
+
+        # Check flag
+        self.assertTrue(mcmc._first_expansion)
+
+    def test_run(self):
+        """
+        Test multiple MCMC iterations of the sampler on a
+        Multivariate Gaussian.
+        """
+        # Set seed for monitoring
+        np.random.seed(2)
+
+        # Create log pdf
+        log_pdf = pints.toy.GaussianLogPDF([2, 4], [[1, 0], [0, 3]])
+
+        # Create mcmc
+        x0 = np.array([1, 1])
+        mcmc = pints.SliceStepoutMCMC(x0)
+
+        # Run multiple iterations of the sampler
+        chain = []
+        while len(chain) < 100:
+            x = mcmc.ask()
+            fx = log_pdf.evaluateS1(x)[0]
+            sample = mcmc.tell(fx)
+            if sample is not None:
+                chain.append(np.copy(sample))
+
+        # Fit Multivariate Gaussian to chain samples
+        np.mean(chain, axis=0)
```

Comment:
What do these lines do at the end of the test? Shouldn't we be comparing these means to the actual means?

Reply:
Hi @ben18785, I used to print these to compare the mean and covariance of the fitted Gaussian with the mean and covariance of the target distribution. It served as a guideline while testing to check whether the sampler was working as expected.

Reply:
I'm with Ben on this one! That line just calculates a mean and then doesn't assign the result to anything...

Reply:
Also just noticed the filenames here should be something like `test_mcmc_slice_sampling.py` (all the mcmc method tests start with test_mcmc)

Reply:
Also, the unit tests shouldn't print anything please :-)
So that means no `print(...)`, but also `mcmc.set_log_to_screen(False)`

Reply:
It'd also be great if you could reduce the time it takes for these tests to run!

Reply:
For example, you don't need to test on different problems unless those problems trigger very different parts of the code. Testing how the methods behaves on different problems is done in the functional testing - which is a separate problem you don't need to worry about :D